src/Microsoft.ML.Onnx/ProtabilityCatalog.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Model.Onnx;
+using Microsoft.ML.UniversalModelFormat.Onnx;
+
+namespace Microsoft.ML
+{
+    public static class ProtabilityCatalog
+    {
+        /// <summary>
+        /// Convert the specified <see cref="ITransformer"/> to ONNX format. Note that ONNX uses Google's Protobuf so the returned value is a Protobuf object.
+        /// </summary>
+        /// <param name="catalog">A field in <see cref="MLContext"/> which this function associated with.</param>
+        /// <param name="transform">The <see cref="ITransformer"/> that will be converted into ONNX format.</param>
+        /// <param name="inputData">The input of the specified transform.</param>
+        /// <returns></returns>
+        public static ModelProto ConvertToOnnx(this ModelOperationsCatalog.PortabilityTransforms catalog, ITransformer transform, IDataView inputData)
+        {
+            var env = new MLContext(seed: 1);
```

Comment:
>var env = new MLContext(seed: 1); [](start = 12, length = 33)

Sorry, why are we creating the environment, instead of getting it from the catalog?

Reply:
I will make `ModelOperationsCatalog.Environment` a BestFriend and access it from here (another assembly).

---
In reply to: [245427576](https://github.com/dotnet/machinelearning/pull/2013#discussion_r245427576) [](ancestors = 245427576)